corehq/apps/cloudcare/static/cloudcare/js/formplayer/menus/views/query.js

```diff
@@ -460,10 +496,80 @@ hqDefine("cloudcare/js/formplayer/menus/views/query", function () {
 
     });
 
+    var GroupedQueryView = Marionette.CollectionView.extend({
+        tagName: "tr",
+        template: _.template($("#query-view-group-template").html() || ""),
+        childView: QueryView,
+        childViewContainer: "#query-group-content",
+
+        childViewOptions: function () {
+            return {parentView: this.options.parentView};
+        },
+
+        ui: {
+            groupHeader: '.search-query-group-header',
+        },
+
+        events: {
+            'click @ui.groupHeader': 'updateArrow',
+        },
+
+        updateArrow: function (e) {
+            const arrow = $(e.currentTarget).children('i');
+            if (arrow.hasClass('fa-angle-double-down')) {
+                arrow.removeClass('fa-angle-double-down');
+                arrow.addClass('fa-angle-double-up');
+            } else {
+                arrow.removeClass('fa-angle-double-up');
+                arrow.addClass('fa-angle-double-down');
+            }
+        },
+
+        templateContext: function () {
+            let groupName = this.options.groupName === undefined ?
+                "" : markdown.render(this.options.groupName.trim());
+            return {
+                groupName: DOMPurify.sanitize(groupName),
+                groupKey: this.options.groupKey,
+                required: this.options.required,
+                named: this.options.groupName.length > 0,
+            };
+        },
+    });
+
     var QueryListView = Marionette.CollectionView.extend({
         tagName: "div",
         template: _.template($("#query-view-list-template").html() || ""),
-        childView: QueryView,
+
+        childView(item) {
+            if (item.has("groupName")) {
+                return GroupedQueryView;
+            } else {
+                return QueryView;
+            }
+        },
+
+        buildChildView(child, ChildViewClass, childViewOptions) {
+            let options = {};
+
+            if (child.has("groupName")) {
```

Comment:
No. that is to identify if the child is a group as set in `groupDisplays`

Reply:
But if a group's `groupName` is an empty string, then wouldn't this incorrectly evaluate as false? 

If we check `groupKey`'s value instead, then when it's null, it wouldn't be a group. Otherwise, if `groupKey` contains a value, it is a group.

Reply:
This is to distinguish between showing groups or not. If `groupHeaders` is empty the `displays` will be used as they are (and they do not have a field `groupName`. If there are group headers the `collection` will be "group" objects that have a `groupName` field

Reply:
oh i see, you're right. I mistook that it was checking the value of `groupName` but it's just checking if the property exists.